src/coreclr/src/jit/codegenxarch.cpp

```diff
@@ -2989,13 +2989,19 @@ void CodeGen::genCodeForInitBlkUnroll(GenTreeBlk* node)
     // Fill as much as possible using SSE2 stores.
     if (size >= XMM_REGSIZE_BYTES)
     {
+#ifdef FEATURE_SIMD
+        bool useYmm = (compiler->getSIMDVectorRegisterByteLength() == YMM_REGSIZE_BYTES) && (size >= YMM_REGSIZE_BYTES);
```

Comment:
Do we want to tie this to the size of `Vector<T>` rather than to the underlying ISAs that are available (based on the instructions used)?

Reply:
Do you mean to replace `compiler->getSIMDVectorRegisterByteLength() == YMM_REGSIZE_BYTES` with `compiler->>getSIMDSupportLevel() == SIMD_AVX2_Supported`? I do not have a preference here, what is your opinion?

Reply:
No, to replace it with a `compSupports(InstructionSet_AVX2)` check so we aren't tied to `Vector<T>` when we aren't actually using `Vector<T>`.

The appropriate check might change slightly with https://github.com/dotnet/runtime/pull/33274, but I imagine this PR would get merged first. CC. @davidwrighton

Reply:
Got it, will do that before merge, thanks.